src/v/v8_engine/executor.h

```diff
@@ -0,0 +1,198 @@
+        return _on_done.wait().then([this] {
+            if (_state.failed()) {
+                return ss::make_exception_future<>(
+                  std::move(_state).get_exception());
+            } else {
+                return ss::make_ready_future<>();
+            }
+        });
+    }
+
+    Func _func;
+    ss::future_state<bool> _state;
+    condition _on_done;
+
+    std::chrono::milliseconds _timeout{};
+};
+
+} // namespace internal
+
+// This class implement executor thread (std::thread) for runing v8 script. It
+// uses boost::lockfree_queue for storing task.
+class executor {
+public:
+    executor(uint8_t cpu_id);
+
+    executor(const executor& other) = delete;
+    executor& operator=(const executor& other) = delete;
+    executor(executor&& other) = delete;
+    executor& operator=(executor&& other) = delete;
+
+    ~executor();
+
+    // Start ss::sharded for ss::gate and ss::abort_source
+    ss::future<> start();
+
+    // Stop executor. Frst of all we need to stop semaphore for cancel waitnig
+    // tasks. After that we need complete all task in queue and finally stop
+    // std::threads
+    ss::future<> stop();
+
+    bool is_stopping() const;
+
+    // clang-format off
+    CONCEPT(
+    template<typename T>
+    concept FuncFoExecutorConcept = requires(T func) {
+        { func() } -> void;
+        { func.cancel() } -> void;
+        { func.on_end() } -> void;
+    };)
+    // clang-format on
+
+    /// Submit new task in executor.
+    ///
+    /// \param start_func is used for set watchdog
+    /// \param func_for_executor is v8 script
+    template<typename WrapperFuncForExecutor>
+    CONCEPT(requires FuncFoExecutorConcept<WrapperFuncForExecutor>)
+    auto submit(
+      WrapperFuncForExecutor&& func_for_executor,
+      std::chrono::milliseconds timeout) {
+        return ss::with_gate(
+          _gates.local(),
+          [this,
+           func_for_executor = std::forward<WrapperFuncForExecutor>(
+             func_for_executor),
+           timeout]() mutable {
+              auto new_task
+                = std::make_unique<internal::task<WrapperFuncForExecutor>>(
+                  std::forward<WrapperFuncForExecutor>(func_for_executor),
+                  timeout);
+              auto fut = new_task->get_future();
+              _tasks.push(new_task.get());
+              _has_task_cv.notify_one();
+              return fut.finally([new_task = std::move(new_task)] {});
+          });
+    }
+
+private:
+    void rearm_watchdog(
+      internal::work_item& task, std::chrono::milliseconds timeout);
+
+    void cancel_watchdog(internal::work_item& task);
+
+    // Do we need to pin all thread for one core?
+    void pin(unsigned cpu_id);
+
+    // Main loop for threads in executor
+    void loop();
+
+    std::atomic<bool> _is_stopped{false};
+
+    std::mutex mutex;
+    std::condition_variable _has_task_cv;
+    std::thread _threads;
+
+    ss::sharded<ss::gate> _gates;
```

Comment:
We do not need `sharded` gate. let's do it the same way as with timer

Reply:
Done